include/ruby/internal/intern/thread.h

```diff
@@ -73,7 +73,8 @@ VALUE rb_mutex_synchronize(VALUE mutex, VALUE (*func)(VALUE arg), VALUE arg);
 
 VALUE rb_thread_scheduler_get(VALUE);
 VALUE rb_thread_scheduler_set(VALUE, VALUE);
-VALUE rb_current_thread_scheduler(void);
+
+VALUE rb_thread_scheduler_if_nonblocking(VALUE thread);
```

Comment:
`if_nonblocking` is ambiguous so I don't like this name. However, if this function is not a public (not in include/ directory), it is acceptable because we can change it later.

Also `rb_current_thread_scheduler(void)` like API (return current fiber scheduler, with GVL) is better because most of case it called with `rb_thread_scheduler_if_nonblocking(rb_thread_current()`.

Reply:
"ambiguous" means thread also has blocking status.